core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -433,7 +435,7 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
     this.consumers.head.commitSync(Map(tp -> new OffsetAndMetadata(5)).asJava)
   }
 
-  @Test(expected = classOf[TopicAuthorizationException])
+  @Test(expected = classOf[KafkaException])
```

Comment:
I think an UNKNOWN_TOPIC_OR_PARTITION error in commit can probably be treated as a fatal error which we raise to the user. If the topic doesn't exist, then you shouldn't have any offsets to commit anyway. Maybe we can do the following:

1. Update the possible error codes in `OffsetCommitResponse` and the protocol documentation to include this error code.
2. Catch the error explicitly in `OffsetCommitResponseHandler` and rethrow either as a `UnknownTopicOrPartitionException` with a better message or `KafkaException` as is currently done. At least then the expected behavior is clearly implemented.

I'm wondering if it makes sense to do similarly for the OffsetFetch API. We may actually prefer in that case to retry since the topic may be created automatically by a pending TopicMetadata request. That said, it may be better to keep the current behavior for the sake of this patch. Also, since no one is complaining about it that I've seen, maybe we shouldn't "fix" it to be consistent with behavior that we want to deprecate anyway. In that case, we should do 1. and 2. similarly for offset fetches. Thoughts?